### DISCUSSION TITLE
FatalError on attempt to create folder with ShellExec Custom Action

### DISCUSSION THREAD

**Discussion by CadMan77** [wixtoolset/issues#7586] [Questions]
I'm playing around with concept of Custom Actions.
I've created two CAs:
   `<CustomAction Id="SetCmdLine_TempDirCreate" Property="WixShellExecTarget" Value="mkdir C:\temp\a1" />`
   `<CustomAction Id="CreateTempDir" BinaryKey="WixCA" DllEntry="WixShellExec" Impersonate="yes" />`

I then call them consequently by custom PushButton:
   `<Publish Event="DoAction" Value="SetCmdLine_TempDirCreate">1</Publish>`
   `<Publish Event="DoAction" Value="CreateTempDir">1</Publish>`

After pressing the button I get "...Setup Wizard ended prematurely" message.
Installer log file contains FatalError on CreateTempDir.

**What I'm missing?**

PS:
I do know that I can create folder by means of Directory Id.

**Comment by robmen** [comment]
My guess is that `mkdir` isn't an executable but a command built into `cmd.exe`. Thus you can't shell execute it, but would need to shell execute cmd.exe to then run the `mkdir` command.

This is all a horrible idea by the way. Don't do this.
